### PR TITLE
fix jwt flow bugs

### PR DESCRIPTION
**Pull request by huminzhi**

### CODE REVIEW

**Comment by huminzhi** [comment]
@kchitalia @helenren @bhariharan This is to fix some issue discovered

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java [hunk 1]
```diff
@@ -289,6 +289,9 @@ public LoginOptions getLoginOptions(String jwt, String url) {
                 loginOptions = new LoginOptions(url, getPasscodeHash(), config.getOauthRedirectURI(),
                         config.getRemoteAccessConsumerKey(), config.getOauthScopes(), null, jwt);
             }
+        } else {
```
Comment (huminzhi) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/app/SalesforceSDKManager.java]: so this is to fix a scenario where the app is first launched normally, then background and foreground through the link, we are not updating loginOption

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java [hunk 2]
```diff
@@ -406,10 +406,13 @@ protected TokenEndpointResponse performRequest(LoginOptions options) {
         @Override
         protected void onPostExecute(TokenEndpointResponse tr) {
             if (tr != null && tr.authToken != null) {
-                loginOptions.jwt = tr.authToken;
+                loginOptions.setJwt(tr.authToken);
                 doLoadPage(true);
             }
-            loginOptions.jwt = null;
+            else {
```
Comment (huminzhi) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java]: This is to fix a scenario where if the jwt expired, the app will stuck at a blank screen

**Comment by bhariharan** [comment]
👍

**Pull request merged**